modules/liana.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  options.services.lianad = {
+    enable = mkEnableOption "lianad bitcoin wallet";
+    daemon = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc "Whether to run the process as a UNIX daemon (double fork magic).";
+    };
+    data_dir = mkOption {
+      type = types.path;
+      default = "/var/lib/lianad";
+      description = mdDoc "Path to the folder where we should store the application data.";
+    };
+    main_descriptor = mkOption {
+      type = types.str;
+      default = "wsh(or_d(pk([0dd8c6f0/48'/1'/0'/2']tpubDFMbZ7U5k5hEfsttnZTKMmwrGMHnqUGxhShsvBjHimXBpmAp5KmxpyGsLx2toCaQgYq5TipBLhTUtA2pRSB9b14m5KwSohTDoCHkk1EnqtZ/<0;1>/*),and_v(v:pkh([d4ab66f1/48'/1'/0'/2']tpubDEXYN145WM4rVKtcWpySBYiVQ229pmrnyAGJT14BBh2QJr7ABJswchDicZfFaauLyXhDad1nCoCZQEwAW87JPotP93ykC9WJvoASnBjYBxW/<0;1>/*),older(65535))))#7nvn6ssc";
+      description = mdDoc "The wallet descriptor.";
```

Comment:
Isn't it better to move these hardcoded defaults to `example` instead of `default`.
I can see easily an user shooting himself in the foot.

Reply:
ideally this is a `systemd` parameter

the problem is that `lianad`'s CLI is still very limited, so everything needs to be written indo a `config.toml`

so the challenge here is:
- declare all options from [`lianad_config_example.toml`](https://github.com/wizardsardine/liana/blob/master/contrib/lianad_config_example.toml) so they're available as `systemd` configurations (which seems to be the most common pattern)
- 

I started doing a [heredoc inside the `preStart`](
https://github.com/plebhash/nix-bitcoin/blob/e69002dc223bd34f538ff6594e72663fc0109ecb/modules/liana.nix#L61) but still unsure whether that is the best approach.

Reply:
Yes but still in a footgun for the user.
Check my comment below on how to interpolate TOML file contents into a string.